google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelector.java

```diff
@@ -167,6 +168,18 @@ public void componentHidden(ComponentEvent event) {
           }
         });
 
+    getTextField()
+        .getDocument()
+        .addDocumentListener(
+            new DocumentAdapter() {
+              @Override
+              protected void textChanged(DocumentEvent e) {
+                if (popupPanel != null) {
+                  popupPanel.setFilter(getText());
```

Comment:
In the latest iteration of this, hows the delay on various systems? I think its important that we validate that it responds quickly enough (<10ms or whatever so as not to be visible / annoying to the user).

Reply:
Just to add my initial impression after giving it a test drive

On my linux box it feels pretty snappy overall. However, typing in the first letter has a noticeable delay. Subsequent letters though have a negligible delay. Could that initial delay be optimized?

Reply:
I was able to get it into a state where the delay on the first letter keeps increasing (exponentially it seems):

Each time I type a letter into the selector field, backspace, then type it again the delay gets longer. Eventually it gets to the point even of crashing the IDE. Is there maybe some state that is being held onto mistakenly?

Reply:
So there's no explicitly set delay; any delay you're experiencing is it applying the filter to every project, telling the tree there was a model update, and redrawing all of the matching projects.

Out of curiosity, how many projects do you have? Do you have multiple accounts setup too? I tried typing the first letter over and over again for a few minutes with my 300+ projects and didn't notice any sort of build-up of delay.

Reply:
I don't have more than 300+ projects and have it set up with only one account.

I noticed that I was able to trigger this delay by playing first typing in the selector, clicking off, then returning to it. I don't really have exact reproduction steps.

Reply:
Fixed it; there was a memory leak on the `popupPanel`. PTAL